hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/HapiTestEngine.java

```diff
@@ -243,33 +250,67 @@ public SkipResult shouldBeSkipped(HapiTestEngineExecutionContext context) {
+
         /** The method under test */
         private final Method testMethod;
 
+        private final Set<TestTag> testTags;
+
         public MethodTestDescriptor(Method testMethod, ClassTestDescriptor parent) {
             super(
                     parent.getUniqueId().append("method", testMethod.getName()),
                     testMethod.getName(),
                     MethodSource.from(testMethod));
             this.testMethod = testMethod;
+            this.testTags = getTagsIfAny(testMethod);
             setParent(parent);
         }
 
+        private Set<TestTag> getTagsIfAny(Method testMethod) {
+            // When a method has a single @Tag annotation, we retrieve it by filtering for Tag.class.
+            // In cases where a method has multiple @Tag annotations, we use Tags.class to access all of them.
+            // Ideally, Tags.class should encompass both single and multiple @Tag annotations,
+            // but the current implementation does not support this.
+            final var tagsAnnotation = testMethod.getAnnotation(Tags.class);
+            final var tagAnnotation = testMethod.getAnnotation(Tag.class);
+
+            final var tags = new HashSet<TestTag>();
+            if (tagsAnnotation != null) {
+                tags.addAll(Arrays.stream(tagsAnnotation.value())
+                        .map(t -> TestTag.create(t.value()))
+                        .toList());
+            } else if (tagAnnotation != null) {
+                tags.add(TestTag.create(tagAnnotation.value()));
+            }
+            return tags;
+        }
+
         @Override
         public Type getType() {
             return Type.TEST;
         }
 
         @Override
         public SkipResult shouldBeSkipped(HapiTestEngineExecutionContext context) {
-            final var annotation = AnnotationSupport.findAnnotation(testMethod, Disabled.class);
-            if (!AnnotationSupport.isAnnotated(testMethod, HapiTest.class)) {
+            final var isHapiTestAnnotated = AnnotationSupport.isAnnotated(testMethod, HapiTest.class);
+            final var disabledAnnotation = AnnotationSupport.findAnnotation(testMethod, Disabled.class);
+
+            if (!isHapiTestAnnotated) {
                 return SkipResult.skip(testMethod.getName() + " No @HapiTest annotation");
-            } else if (annotation.isPresent()) {
-                final var msg = annotation.get().value();
+            } else if (disabledAnnotation.isPresent()) {
+                final var msg = disabledAnnotation.get().value();
                 return SkipResult.skip(msg == null || msg.isBlank() ? "Disabled" : msg);
             }
-            return SkipResult.doNotSkip();
+
+            final var tagFilter = System.getenv("TAG_FILTER");
+            TestTag tag = tagFilter != null ? TestTag.create(tagFilter) : null;
+            if (tag == null && testTags.isEmpty()) {
+                return SkipResult.doNotSkip();
+            } else if (testTags.contains(tag)) {
```

Comment:
I think we will need to support more complex logic. At a minimum, we need to be able to specify "run all tests except the ones that are marked as slow". Ideally we support standard expressions as defined in the [documentation](https://junit.org/junit5/docs/current/user-guide/#running-tests-tag-expressions)

Maybe we can take advantage of a default implementation like [TagFilter](https://junit.org/junit5/docs/5.2.0/api/org/junit/platform/launcher/TagFilter.html)